ic-cdk/src/call.rs

```diff
@@ -163,7 +98,46 @@ pub enum CallError {
     CandidDecodeFailed(String),
 }
 
+/// The error type for inter-canister calls.
+#[derive(Debug, Clone)]
+pub struct CallRejected {
+    // All fields are private so we will be able to change the implementation without breaking the API.
+    // Once we have `ic0.msg_error_code` system API, we will only store the error_code in this struct.
+    // It will still be possible to get the [`RejectCode`] using the public getter,
+    // because every error_code can map to a [`RejectCode`].
+    reject_code: RejectCode,
+    reject_message: String,
+    sync: bool,
+}
+
+impl CallRejected {
+    /// Returns the [`RejectCode`].
+    pub fn reject_code(&self) -> RejectCode {
+        self.reject_code
+    }
+
+    /// Returns the reject message.
+    ///
+    /// When the call was rejected asynchronously (IC rejects the call after it was enqueued),
+    /// this message is set with [`msg_reject`](crate::api::msg_reject).
+    ///
+    /// When the call was rejected synchronously (`ic0.call_preform` returns non-zero code),
+    /// this message is set to a fixed string ("failed to enqueue the call").
```

Comment:
I'd recommend to stick to the current message:
- for the sake of backward-compatibility (no change in behavior)
- "failed to enqueue the call" sounds like the queue is the issue, but the failure could also be due to cycles (for instance)